ophyd/device.py

```diff
@@ -184,12 +189,48 @@ def __get__(self, instance, owner):
             return self
 
         if self.attr not in instance._signals:
-            instance._signals[self.attr] = self.create_component(instance)
+            cpt_inst = self.create_component(instance)
+            instance._signals[self.attr] = cpt_inst
+
+            for event_type, functions in self._subscriptions.items():
+                for func in functions:
+                    cpt_inst.subscribe(types.MethodType(func, instance),
+                                       event_type=event_type,
+                                       run=cpt_inst.connected)
 
         return instance._signals[self.attr]
 
     def __set__(self, instance, owner):
-        raise RuntimeError('Use .put()')
+        raise RuntimeError('Do not use setattr with components; use '
+                           'cpt.put(value)')
+
+    def subscriptions(self, event_type):
+        '''(Decorator) Specify subscriptions callbacks in the Device definition
+        '''
+        if isinstance(self, DynamicDeviceComponent):
+            raise NotImplementedError('DynamicDeviceComponent does not yet '
+                                      'support decorator subscriptions')
```

Comment:
perhaps this should be moved to the DynamicDeviceComponent class rather than checking here

Reply:
👍 Good call